broker-core/src/test/java/org/apache/qpid/server/security/encryption/AESGCMKeyFileEncrypterTest.java

```diff
@@ -0,0 +1,364 @@
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.google.common.util.concurrent.SettableFuture;
+import org.junit.Before;
+import org.junit.Test;
+
+import org.apache.qpid.server.SystemLauncher;
+import org.apache.qpid.server.SystemLauncherListener;
+import org.apache.qpid.server.model.AuthenticationProvider;
+import org.apache.qpid.server.model.Broker;
+import org.apache.qpid.server.model.BrokerModel;
+import org.apache.qpid.server.model.ConfiguredObject;
+import org.apache.qpid.server.model.JsonSystemConfigImpl;
+import org.apache.qpid.server.model.SystemConfig;
+import org.apache.qpid.server.model.User;
+import org.apache.qpid.server.util.FileUtils;
+import org.apache.qpid.test.utils.UnitTestBase;
+
+public class AESGCMKeyFileEncrypterTest extends UnitTestBase
+{
+    private static final String PLAIN_SECRET = "secret";
+    private final SecureRandom _random = new SecureRandom();
+    public static final String PLAINTEXT = "guestkey";
+    private Path _configurationLocation;
+    private Path _workDir;
+    private Broker<?> _broker;
+    private SystemLauncher _systemLauncher;
+    private Map<String, Object> attributes = new HashMap<>();
+
+    private void createBrokerWithEncrypter(final Object encrptionType) throws Exception
+    {
+        try
+        {
+            _workDir = Files.createTempDirectory("qpid_work_dir");
+            _configurationLocation = Files.createTempFile(_workDir, "config", ".json");
+            Map<String, Object> context = new HashMap<>();
+            context.put("qpid.work_dir", _workDir.toFile().getAbsolutePath());
+            Map<String, Object> config = new HashMap<>();
+            config.put(ConfiguredObject.NAME, getTestName());
+            config.put(Broker.MODEL_VERSION, BrokerModel.MODEL_VERSION);
+            config.put(Broker.CONFIDENTIAL_CONFIGURATION_ENCRYPTION_PROVIDER, encrptionType);
+            new ObjectMapper().writeValue(_configurationLocation.toFile(), config);
+
+            attributes.put("storePath", _configurationLocation.toFile().getAbsolutePath());
+            attributes.put("preferenceStoreAttributes", "{\"type\": \"Noop\"}");
+            attributes.put(SystemConfig.TYPE, JsonSystemConfigImpl.SYSTEM_CONFIG_TYPE);
+            attributes.put(SystemConfig.STARTUP_LOGGED_TO_SYSTEM_OUT, Boolean.FALSE);
+            attributes.put(SystemConfig.CONTEXT, context);
+            final SettableFuture<SystemConfig<?>> configFuture = SettableFuture.create();
+            _systemLauncher = new SystemLauncher(new SystemLauncherListener.DefaultSystemLauncherListener()
+            {
+                @Override
+                public void onContainerResolve(final SystemConfig<?> systemConfig)
+                {
+                    configFuture.set(systemConfig);
+                }
+            });
+            _systemLauncher.startup(attributes);
+            final SystemConfig<?> systemConfig = configFuture.get(10, TimeUnit.SECONDS);
```

Comment:
10 is magic number... can it be moved into a constant

Reply:
Done!